tests/icub-model-test.cpp

```diff
@@ -259,6 +259,84 @@ bool checkAxisDirections(iDynTree::KinDynComputations & comp)
+    "neck_yaw", "r_shoulder_roll",
+    "r_shoulder_yaw", "r_elbow",
+    "r_wrist_prosup", "r_wrist_pitch",
+    "r_wrist_yaw", "l_shoulder_roll",
+    "l_shoulder_yaw","l_elbow",
+    "l_wrist_prosup", "l_wrist_pitch",
+     "l_wrist_yaw"};
+
+    std::vector<iDynTree::Direction> expectedDirectionInRootLink;
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(-1,0,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0,-1,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0,-1,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(-1,0,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0,0,-1));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0,-1,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0,1,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(-1,0,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(1,0,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0,0,1));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0,-1,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0,1,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(1,0,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0,1,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0,0,-1));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0.250563,0.935113,0.250563));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(-0.250563,0.935113,-0.250563));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0,-1,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(1,0,0));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(-1.62555e-21,-1.1e-15,1));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(-0.961047,-0.271447,-0.0520081));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(-0.0713662,0.0619303,0.995526));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0.267012,-0.960459,0.0788901));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(-0.0713662,0.0619303,0.995526));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0.961047,0.271447,0.0520081));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(-0.267012,0.960459,-0.0788901));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0.961047,-0.271447,0.0520081));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0.0713662,0.0619303,-0.995526));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(-0.267012,-0.960459,-0.0788901));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0.0713662,0.0619303,-0.995526));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(-0.961047,0.271447,-0.0520081));
+    expectedDirectionInRootLink.push_back(iDynTree::Direction(0.267012,0.960459,0.0788901));
```

Comment:
Sorry for being picky, but the code as written in this form would be quite complicate to maintain, as it is difficult to understand which axis correspond to which joint. 

Could it make sense to write it as: 
~~~
    axisNames.push_back("torso_roll");
    expectedDirectionInRootLink.push_back(iDynTree::Direction(-1,0,0));
    axisNames.push_back("l_hip_pitch");
    expectedDirectionInRootLink.push_back(iDynTree::Direction(0,-1,0));
    axisNames.push_back("r_hip_pitch");
    expectedDirectionInRootLink.push_back(iDynTree::Direction(0,-1,0));
~~~

And so on so forth?

Reply:
Now should be clear, it would be better to creat a struct with name and expected rotation but it may be refactored in the future